### PR TITLE
Custom event hook to enable custom handling of DataMarkers in structure files

### PR DESCRIPTION
**Pull request by PssbleTrngle**
This enables mod to spawn mobs on generation of vanilla structures, or prevent them

### CODE REVIEW

File: src/main/java/net/minecraftforge/event/world/StructureDataMarkerEvent.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@Cancelable
+public class StructureDataMarkerEvent extends Event
+{
+
+    public final String function;
+    public final BlockPos pos;
+    public final IWorld world;
+    public final Random rand;
+    public final MutableBoundingBox sbb;
+    public final TemplateStructurePiece structurePiece;
```
Comment (TheSilkMiner) [inline src/main/java/net/minecraftforge/event/world/StructureDataMarkerEvent.java]: Don't expose fields, rather provide Getters and setters if needed

File: src/main/java/net/minecraftforge/event/world/StructureDataMarkerEvent.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import net.minecraft.util.math.MutableBoundingBox;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.gen.feature.structure.TemplateStructurePiece;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.Random;
+
+@Cancelable
+public class StructureDataMarkerEvent extends Event
```
Comment (TheSilkMiner) [inline src/main/java/net/minecraftforge/event/world/StructureDataMarkerEvent.java:22]: Missing documentation

**Comment by Cyborgmas** [comment]
You changed a bunch of imports for wild cards, pretty sure you should change them back

**Comment by PssbleTrngle** [comment]
You are right, looks like the IDE did that

**Comment by sciwhiz12** [comment]
I am closing this PR as stale. If you or anyone wishes to continue this PR, please update the PR to `1.18.x` first. 
Thank you for your contributions. 👋

**Pull request closed without merging**